tests/acceptance/stepDefinitions/publicLinkContext.js

```diff
@@ -5,47 +5,149 @@ const sharingHelper = require('../helpers/sharingHelper')
-    return client.page.FilesPageElement
+    await client.page.FilesPageElement
       .filesList()
       .closeSidebar(100)
       .openPublicLinkDialog(resource)
-      .addNewLink(settings.role)
+    return client.page.FilesPageElement.publicLinksDialog().addNewLink(settings)
   }
 )
 
 When('the public uses the webUI to access the last public link created by user {string}', async function (linkCreator) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
   return client.page.publicLinkFilesPage().navigateAndWaitTillLoaded(lastShareToken)
 })
 
-When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
-  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
+When('the public uses the webUI to access the last public link created by user {string} with password {string}',
+  async function (linkCreator, password) {
+    const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+    const lastShareToken = lastShare.token
+    await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
+    return client.page.publicLinkPasswordPage().submitPublicLinkPassword(password)
+  }
+)
 
-  return client.page.publicLinkPasswordPage().submitPublicLinkPassword(password)
+When('last public share of the user {string} is deleted using API', async function (sharer) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(sharer)
+  const lastShareId = lastShare.id
+  return sharingHelper.deleteLastPublicLinkShare(sharer, lastShareId)
+})
+
+Then('user {string} should not have any public link',
+  async function (sharer) {
+    const resp = await sharingHelper.getAllPublicLinkShares(sharer)
+    assert.deepStrictEqual(
+      resp.length, 0, 'User has shares. Response: ' + resp)
+  })
+
+Then('the fields of the last public link share response of user {string} should include',
+  function (linkCreator, dataTable) {
+    const fieldsData = dataTable.rowsHash()
+    return sharingHelper.assertUserLastPublicShareDetails(linkCreator, fieldsData)
+  })
+
+Then('the user should see an error message on the public link share dialog saying {string}',
+  function (errorMessage) {
+    return client.page.FilesPageElement
+      .publicLinksDialog()
+      .assertErrorMessageForPublicLinks(errorMessage)
+  })
+
+Then('as user {string} the folder {string} should not have any public link', async function (sharer, resource) {
+  const publicLinkShares = await sharingHelper.getAllPublicLinkShares(sharer)
+  let found = false
+  for (const share of publicLinkShares) {
+    if (share.path === '/' + resource && share.share_type === 3 && share.uid_owner === sharer) {
```

Comment:
Also, make it work for paths starting with `/` or without it.
```suggestion
    if (share.path === '/' + resource && share.share_type === SHARE_TYPES.public_link) {
```